### ISSUE TITLE
[1.12.2] EntityCitizen (AiSleep) Bug

### ISSUE THREAD

**Issue by demfels** [ldtteam/minecolonies#2798]
**Crash Log: https://pastebin.com/767TgemX**

### Minecolonies version

version 1.12.2-0.9.87-ALPHA-universal.jar

### Expected behavior

- Sleep through night
- Villagers Wake
- Game does not crash

### Actual behaviour

- Sleep through night
- Villagers Wake
- Game Crashes

### Steps to reproduce the problem

Seems to be the same or similar to**https://github.com/ldtteam/minecolonies/issues/2228**

**Comment by demfels** [comment]
I was able to track down the source, based on a location in the error log.  It looks like one of my players is upgrading a tier 5 home, and the beds are destroyed.  I suspect this is the cause of the issue, like the issue#2228 says, potentially not handled error here.


Also, we just removed the citizens from the problem hut as a work around and that handled the issue for us until he completed the building.
